mmpose/datasets/pipelines/bottom_up_transform.py

```diff
@@ -572,38 +708,138 @@ def __call__(self, results):
 class BottomUpGenerateHeatmapTarget:
     """Generate multi-scale heatmap target for bottom-up.
 
+    Required key: 'joints', 'mask' and 'center'.
+
+    Modifies key: 'target', 'heatmaps' and 'masks'.
+
     Args:
-        sigma (int): Sigma of heatmap Gaussian
-        max_num_people (int): Maximum number of people in an image
+        sigma (int or tuple): Sigma of heatmap Gaussian. If sigma is a tuple,
+            the first item should be the sigma of keypoints and the second
+            item should be the sigma of center.
+        bg_weight (float): Weight for background. Default: 1.0.
+        gen_center_heatmap (bool): Whether to generate heatmaps for instance
+            centers. Default: False.
         use_udp (bool): To use unbiased data processing.
             Paper ref: Huang et al. The Devil is in the Details: Delving into
             Unbiased Data Processing for Human Pose Estimation (CVPR 2020).
     """
 
-    def __init__(self, sigma, use_udp=False):
+    def __init__(self,
+                 sigma,
+                 bg_weight=1.0,
+                 gen_center_heatmap=False,
+                 use_udp=False):
+
+        if isinstance(sigma, int):
+            sigma = (sigma, )
+        if gen_center_heatmap:
+            assert len(sigma) == 2, 'sigma for centers must be given if ' \
+                                    '`gen_center_heatmap` is True. ' \
+                                    'e.g. sigma=(2, 4)'
+
         self.sigma = sigma
+        self.bg_weight = bg_weight
+        self.gen_center_heatmap = gen_center_heatmap
         self.use_udp = use_udp
 
-    def _generate(self, num_joints, heatmap_size):
+    def _generate(self, num_joints, sigma, heatmap_size):
```

Comment:
is sigma necessary?

Reply:
`BottomUpGenerateHeatmapTarget` is modified to support the generation of both keypoint and instance center heatmaps. These two kinds of heatmaps are generally rendered with different sigma. 

Thus, `self.sigma` is changed to a tuple that contains sigmas for keypoint and instance center heatmaps. The parameter `sigma` of function `_generate` is the sigma for either keypoint or instance center heatmaps